File: code_snippets/python/src/use_cases/workflows/submit.py
import requests

# <start_here>
restate = "http://localhost:8080"

# 1. Submit the workflow
user_id = "myUser123"
# !mark
requests.post(
    # !mark
    f"${restate}/signupWorkflow/${user_id}/run/send",
    json={"email": "user@user.com", "name": "Pete"},
    headers={"Content-Type": "application/json"},
)

# Do something else, with workflow running in the background

# 2. Attach back to the workflow
# !mark
response = requests.get(
    # !mark
    f"${restate}/restate/workflow/signupWorkflow/${user_id}/attach"
    # !mark
)
# <end_here>

File: code_snippets/ts/src/use_cases/workflows/submit.ts
import * as clients from "@restatedev/restate-sdk-clients";
import { SignUpWorkflow } from "./signup_workflow";

async function submit(user: {id: string, name: string; email: string }) {
  // <start_here>
  // import * as clients from "@restatedev/restate-sdk-clients";
  const rs = clients.connect({ url: "http://localhost:8080" });
  // !mark
  await rs
    // !mark
    .workflowClient<SignUpWorkflow>({ name: "user-signup" }, user.id)
    // !mark
    .workflowSubmit(user);

  // do something else, with workflow running in the background

  // attach back to the workflow
  // !mark
  const result = await rs
    // !mark
    .workflowClient<SignUpWorkflow>({ name: "user-signup" }, user.id)
    // !mark
    .workflowAttach();
  // <end_here>
}
